File: java/TwoSum.java
// https://leetcode.com/problems/two-sum/description/
import java.util.Arrays;

class TwoSum {
    public static void main(String[] args) {
        int[] solution = solution(new int[] { 2, 7, 11, 15 }, 9);
        System.out.println(Arrays.toString(solution));
    }

    public static int[] solution(int[] nums, int target) {
        int[] solution = new int[2];
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    solution[0] = i;
                    solution[1] = j;
                    break;
                }
            }
        }
        return solution;
    }
}

File: ruby/two-sum.rb
def two_sum(nums, target)
  result = []
  nums.each_with_index do |n, i|
    for j in i..(nums.length - 1)
      check = j > i && nums[i] + nums[j] == target
      if check
        result[0] = i
        result[1] = j
        return result
      end
    end
  end
  result
end

print two_sum([3, 2, 4], 6)
